pkg/cloudevents/transport/http/transport.go

```diff
@@ -171,26 +176,28 @@ func (t *Transport) StartReceiver(ctx context.Context) error {
 		}
 		t.realPort = listener.Addr().(*net.TCPAddr).Port
 
-		go func() {
-			log.Print(t.server.Serve(listener))
-			t.server = nil
+		// Shutdown
+		defer func() {
 			t.realPort = 0
+			t.server.Close()
+			t.server = nil
 		}()
-		return nil
-	}
-	return fmt.Errorf("http server already started")
-}
 
-// This blocks until all active connections are closed.
-func (t *Transport) StopReceiver(ctx context.Context) error {
-	if t.server != nil {
-		err := t.server.Close()
-		t.server = nil
-		if err != nil {
-			return fmt.Errorf("failed to close http server, %s", err.Error())
+		errChan := make(chan error, 1)
+		go func() {
+			errChan <- t.server.Serve(listener)
+		}()
+
+		// wait for the server to return or ctx.Done().
+		select {
+		case <-ctx.Done():
+			return nil
+		case err := <-errChan:
+			return err
 		}
+
 	}
-	return nil
+	return fmt.Errorf("http server already started")
```

Comment:
Now that there's a mutex around the entire (blocking) method, can this ever be reached?

Reply:
nope!